src/test_utils/eoa.rs

```diff
@@ -88,6 +89,14 @@ impl<P: Provider + Send + Sync> KakarotEOA<P> {
         let chain_id = self.eth_provider.chain_id().await?.unwrap_or_default();
 
         let bytecode = <KakarotEvmContract as EvmContract>::load_contract_bytecode(contract_name)?;
+        let expected_address = {
+            let expected_eth_address = get_contract_address(
+                ethers::types::Address::from_slice(self.evm_address().expect("EOA should have evm address").as_slice()),
+                ethers::types::U256::from(nonce),
+            );
+            FieldElement::from_byte_slice_be(expected_eth_address.as_bytes())
+                .expect("Failed to convert address to field element")
+        };
```

Comment:
Can you check: I think you can call `self.evm_address().create(nonce)`.
https://docs.rs/alloy-primitives/latest/alloy_primitives/struct.Address.html#method.create